packages/shared/src/lib/cores.ts

```diff
@@ -1,10 +1,22 @@
+import { isNullOrUndefined } from './func';
 import type { LoggedUser } from './user';
 import { CoresRole } from './user';
 
 export const verifyCoresRole = (
   user: LoggedUser,
   coresRole: CoresRole,
-): boolean => user?.coresRole === coresRole;
+): boolean => {
+  if (isNullOrUndefined(user?.coresRole)) {
+    return false;
+  }
 
-export const hasAccessToCores = (user: LoggedUser): boolean =>
-  !verifyCoresRole(user, CoresRole.None);
+  return user?.coresRole === coresRole;
```

Comment:
Technically `user?.coresRole === coresRole` would already return undefined if `user.coresRole` is undefined, but I like being specific